### PR TITLE
created settings sidebar

### LINKED ISSUE #417: migrate(react/next): settings:sidebar
Create a sidebar for settings page similar to nuxt application

### PR DESCRIPTION
**Pull request by Ankur1493**
## What does this PR do?

Created the sidebar for settings page, completed the functionality, some pages are not yet created so will give 404, otherwise it's working here are some ss 
Screenshot from 2024-05-04 18-31-10
Screenshot from 2024-05-04 18-31-43

Fixes #417 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [x] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Tested my code in a local environment
- [x] Commented on my code in hard-to-understand areas
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 1]
```diff
@@ -1,11 +1,211 @@
...
+  useEffect(() => {
+    const segments = pathname.split('/').filter(Boolean);
+    if (segments.length > 0) {
+      let endpoint = segments[segments.length - 1];
+      if (endpoint === 'addresses') {
+        endpoint = `${segments[segments.length - 2]}/${segments[segments.length - 1]}`;
+      }
+      setCurrentUrl(endpoint || '');
+    }
+  }, [pathname]);
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:33]: we don't really need this, just use the path url as key

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 2]
```diff
@@ -1,11 +1,211 @@
...
+      <Flex
+        className="flex-col"
+        gap="1">
+        {personalLinks.map(({ key, label, to, icon }) => (
+          <Link
+            key={key}
+            href={to}>
+            <Flex
+              gap="4"
+              className={`${currentUrl === key ? 'dark:bg-gray-10 bg-gray-4' : ''} rounded p-1 pl-2`}>
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:124]: there is a utility function called `cn` at `@/lib/utils` use that for conditional classNames

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 3]
```diff
@@ -1,11 +1,211 @@
...
+            </Flex>
+          </Link>
+        ))}
+      </Flex>
+      <Text
+        className="border-t pt-4"
+        size="5"
+        weight="bold"
+        color="gray">
+        # Organization
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:136]: `#` is not a part of the title, it is a icon button in Nuxt website which copies the org id and user id.

there is a `useGlobalStore` where you can find all of the data you will need, look for the use of this hook to understand how you should use it

**Review by BlankParticle — CHANGES_REQUESTED** [review]

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx (hunk 1 above)
Comment (Ankur1493) [reply apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:33]: ahh, I overcooked didn't thought of this :)

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 4]
```diff
@@ -1,11 +1,211 @@
...
+  Globe,
+  Users,
+  UserRoundPlus,
+  HeartHandshake
+} from 'lucide-react';
 
 export default function SettingsSidebar() {
+  const pathname = usePathname();
+  const [currentUrl, setCurrentUrl] = useState('');
+  const { orgShortCode } = useParams();
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:22]: better to use do this
```ts
const currentOrgShortCode = useGlobalStore((state) => state.currentOrg.shortCode);
```

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx (hunk 3 above)
Comment (Ankur1493) [reply apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx]: didn't get this so just removed the # as of now

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 5]
```diff
@@ -17,50 +19,43 @@ import {
...
-    const segments = pathname.split('/').filter(Boolean);
-    if (segments.length > 0) {
-      let endpoint = segments[segments.length - 1];
-      if (endpoint === 'addresses') {
-        endpoint = `${segments[segments.length - 2]}/${segments[segments.length - 1]}`;
-      }
-      setCurrentUrl(endpoint || '');
-    }
+    setCurrentUrl(pathname);
   }, [pathname]);
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:28]: just use `pathname` directly instead of `currentUrl`

**Review by BlankParticle — CHANGES_REQUESTED** [review]

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 6]
```diff
@@ -1,11 +1,218 @@
...
+        color="gray">
+        Personal
+      </Text>
+      <Flex
+        className="flex-col"
+        gap="1">
+        {personalLinks.map(({ key, label, to, icon }) => (
+          <Link
+            key={key}
+            href={to}>
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:116]: here you can do `key={to} href={to}`

Comment (Ankur1493) [reply apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:116]: but shouldn't it better to have a separate key, improves readability, same goes for pathname

Comment (BlankParticle) [reply apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:116]: there is no need to repeat the props, also you can just rename `pathname` as `currentUrl`

Comment (Ankur1493) [reply apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:116]: Okk, doing it, but I still feel I am right

Comment (BlankParticle) [reply apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:116]: using separate key is good when you don't know what your data might be, for this you are defining your keys yourself, so there is no chance of collision. Have to props named `key` and `to` with same string is not the best thing

**Comment by Ankur1493** [comment]
creating a new pr, this just become a lot

**Pull request closed without merging**